reference-implementation/__tests__/composition.js

```diff
@@ -0,0 +1,367 @@
+'use strict';
+const { URL } = require('url');
+const { parseFromString } = require('../lib/parser.js');
+const { appendMap } = require('../lib/composer.js');
+
+const mapBaseURL = new URL('https://example.com/app/index.html');
+
+function composeMaps(mapLikes) {
```

Comment:
It'd be nice to add a comment explaining that this functions serves a few purposes (as far as I can tell):

- Allowing the input to be plain JS objects, instead of import map objects or strings
- Automatically assumes the input will use mapBaseURL.
- Allowing writing tests that list all inputs, instead of creating the first, then appending xN